File: 2D-World-Simulator-Java/OPP-Java/src/op/JavaProject/Organisms/Animals/Fox.java
package op.JavaProject.Organisms.Animals;

import op.JavaProject.Functionality.BoardPoint;
import op.JavaProject.Functionality.Species;
import op.JavaProject.Game.Animal;
import op.JavaProject.Game.Organism;
import op.JavaProject.Game.World;

import java.awt.*;

public class Fox extends Animal {

    public Fox(World organismWorld, BoardPoint point) {
        super(organismWorld, point);
        this.strength = 3;
        this.initiative = 7;
        this.organismSign = "F";
        this.organismSpecie = Species.FOX;
        this.organismColor = new Color(255, 139, 18);
    }

    @Override
    //performing action for fox
    public void Action() {

        BoardPoint newPoint = GetNearPoint(this.GetOrganismPos(),false);
        Organism newPos = organismWorld.GetOrganismAtPos(newPoint);

        //fox is not moving when on desired position there is organism with which he may lose the fight
        if (newPos != null) {
            if (newPos.GetStrength() < strength) {
                Collision(newPos);
            }
            else if (newPos.GetStrength() == strength) {
                if (newPos.GetWhenBorn() > whenBorn) {
                    Collision(newPos);
                }
            }
            else {
                organismWorld.GetCommentator().AddComment(organismSign + " smells " + newPos.GetOrganismSign() + "'s stink...");
            }
        }
        else {
            MoveToGivenField(newPoint);
        }

    }
}

File: 2D-World-Simulator-Python/OPP-Python/Organisms/Animals/Fox.py
from Game.Animal import Animal
from Game.OrganismStatics import OrganismSpecie


class Fox(Animal):
    def __init__(self, organism_world, point):
        super().__init__(organism_world, point, 3, 7, "F", OrganismSpecie.FOX, "#ff8b12")

    # performing action for fox
    def action(self):
        new_point = self.get_near_point(self.organism_position, False)
        new_pos = self.organism_world.get_organism_at_pos(new_point)

        # fox is not moving when on desired position there is organism with which he may lose the fight
        if new_pos is not None:
            if new_pos.get_strength() < self.strength:
                super().collision(new_pos)
            elif new_pos.get_strength() == self.strength:
                if new_pos.get_when_born() > self.when_born:
                    super().collision(new_pos)
            else:
                self.organism_world.get_world_window().add_comment(self.organism_sign + " smells " +
                                                                   new_pos.get_organism_sign() + "'s stink...")
        else:
            self.move_to_given_field(new_point.get_x(), new_point.get_y())
